pkg/kubernetes/app.js

```diff
@@ -25,11 +25,33 @@ define([
             '$routeParams',
             '$location',
             'kubernetesClient',
-            function($scope, $route, $routeParams, $location, client) {
+            'kubernetesFilter',
+            function($scope, $route, $routeParams, $location, client, filter) {
                 $scope.$route = $route;
                 $scope.$location = $location;
                 $scope.$routeParams = $routeParams;
 
+                $scope.filter = filter;
+                $scope.namespaces = client.select("Namespace");
+                client.track($scope.namespaces);
+                $($scope.namespaces).on("changed", function () {
+                    $scope.$digest();
+                });
+
+                $scope.$on("$routeChangeSuccess", function (event, current, prev) {
+                    var request_namespace = $location.search().namespace;
```

Comment:
The logic for retrieving namespace from ```$search``` is here, but the logic for putting namespace into ```$search``` is in kubernetesFilter. Could we put it in one place? In kubernetesFilter seems like a good place, no?